.github/workflows/ci.yml

```diff
@@ -0,0 +1,122 @@
+name: CI
+
+on: [pull_request, push]
+
+jobs:
+  fmt:
+    name: Rustfmt
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          components: rustfmt
+          profile: minimal
+          override: true
+
+      - name: Run rustfmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          components: clippy
+          profile: minimal
+          override: true
+
+      - name: Run clippy
+        uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: -- -D warnings
+
+  audit:
+    name: Security audit
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/audit-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+  build:
+    name: ${{ matrix.build }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        build: [Linux, macOS, Win32-msvc, Win32-gnu, Win64-msvc, Win64-gnu]
+
+        include:
+          - build: Linux
+            os: ubuntu-latest
+          - build: Linux-nightly
+            os: ubuntu-latest
+            rust: nightly
+          - build: macOS
+            os: macOS-latest
+          - build: Win32-msvc
+            os: windows-latest
+            rust: stable-i686-pc-windows-msvc
+            target: i686-pc-windows-msvc
+          - build: Win32-gnu
+            os: windows-latest
+            rust: stable-i686-pc-windows-gnu
+            target: i686-pc-windows-gnu
+          - build: Win64-msvc
+            os: windows-latest
+            rust: stable-x86_64-pc-windows-msvc
+            target: x86_64-pc-windows-msvc
+          - build: Win64-gnu
+            os: windows-latest
+            rust: stable-x86_64-pc-windows-msvc
+            target: x86_64-pc-windows-msvc
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust || 'stable' }}
+          target: ${{ matrix.target }}
+          profile: minimal
+          override: true
+
+      - name: Build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --examples --all
+
+      - name: Test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --all
+
+      - name: Test invocation
+        if: matrix.rust == 'nightly'
```

Comment:
I think it must be the other way around: we test invocation feature on all platforms, but may skip on nightly (though it doesn't seem to make much sense).

Also, it requires setting the `JAVA_HOME` env variable, hence the job currently fails: https://github.com/DarkEld3r/jni-rs/runs/832279070?check_suite_focus=true#step:6:18

Reply:
https://github.com/actions/setup-java should install Java and set `JAVA_HOME`. I'm going to look at this closely.